src/io.jl

```diff
@@ -11,6 +11,7 @@ Read GeoJSON to a GeoInterface.jl compatible object.
 - `numbertype::DataType=Float32`: Use Float64 when the precision is required.
 """
 function read(io; lazyfc=false, ndim=2, numbertype=Float32)
+    eof(io) && seekstart(io)
```

Comment:
Thanks for the PR!

This also applies to strings and Vector{UInt8} as written now, the code should be abstracted out somewhere else!  Perhaps a separate dispatch of `read` for `::IO` objects?  CI is failing for this reason.

Reply:
Also needs some explanatory comments.

Reply:
Is it common to do such a thing? I've never seen it. It seems more up to the caller that the IOBuffer should not be at the end.

Reply:
> Is it common to do such a thing? I've never seen it. It seems more up to the caller that the IOBuffer should not be at the end.

It's common to download GeoJSON data from web links when using GeoMakie

https://github.com/MakieOrg/GeoMakie.jl/blob/d10a28662457eecbcac0371484f0581c035eb82c/examples/german_lakes.jl#L6-L7

https://github.com/MakieOrg/GeoMakie.jl/blob/d10a28662457eecbcac0371484f0581c035eb82c/examples/specialized/graph_on_usa.jl#L9-L10

Reply:
Yeah, I agree, I meant specifically the `seekstart`. That seems like it is not the responsibility of that function.

Reply:
I agree that it is the responsibility of the caller to ensure the IO is in the usable status.
But check `eof` and `seekstart` will made GeoJSON.jl be user-friendly for beginners, especially for those who are unfamiliar with input/output streams.

Reply:
Yeah I appreciate the effort for user-friendliness, but am afraid that then users will expect other functions reading streams do the same, and possibly end up more confused. To keep the example simple it is also fine to just use `Downloads.download("https://its-live-data.s3.amazonaws.com/datacubes/catalog_v02.json")` to get a temp file path.

Reply:
maybe we can providing an seekstart bool argument and/or code example to facilitate this usage scenario.